### ISSUE TITLE
twisted.words.protocols.irc.IRCClient.say docstring neglects to mention "#" may be prepended to channel name

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3968]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ivank reported|
|-|-|
|Trac ID|trac#3968|
|Type|defect|
|Created|2009-08-15 10:52:16Z|

This is sort of like [#3895](https://github.com/twisted/twisted/issues/3895), except the inverse!

Attachments:

* [3968_words-protocols-irc-IRCClient-say.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d8a/d8ad8dee8df9ae8428a5909e8700cdabe73ced24/3d659c7f85764067d489a9441c2dea0206429eaf.patch) (489 bytes) - added by ivank on 2009-08-15 10:54:40Z - fix
* [words-protocol-irc-v2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/d8a/d8ad8dee8df9ae8428a5909e8700cdabe73ced24/c024eedc05ec3da952583128f20d48f68f6015b1.patch) (6356 bytes) - added by jesstess on 2009-08-25 15:44:12Z - updated patch with additional fixes

<details><summary>Searchable metadata</summary>

```
trac-id__3968 3968
type__defect defect
reporter__ivank ivank
priority__normal normal
milestone__ 
branch__ 
branch_author__ivank__jesstess ivank, jesstess
status__closed closed
resolution__fixed fixed
component__words words
keywords__documentation_easy documentation easy
time__1250333536000000 1250333536000000
changetime__1251218314000000 1251218314000000
version__None None
owner__ 
cc__jesstess cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="ivank's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|ivank commented|
|-|-|

oops, wrong method name

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

Anyone can review this.

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_3"></a>|@jesstess commented|
|-|-|

The ticket is correct, and the original patch applies but has a minor grammar issue. 

In reviewing, I noticed several issues in irc.py related to channel prefixes:

a) Every doc string that mentions prepending says "will to prepended to it", rather than "will be prepended to it", including the original patch.
b) Every check for a prefix is "if channel[0] not in '&#!+':" rather than taking advantage of CHANNEL_PREFIXES
c) The doc strings for joined and left mention only 2 of the 4 valid channel prefixes

Since they're all related, I've submitted an updated patch that addresses these 3 issues as well as updating the say doc string.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @jesstess|
|-|-|

Cool, this is a good set of changes.  A few easy requests:

  1. Some of the modified docstrings don't follow the docstring convention described in the coding standard.  Can you format them so that the opening triple quote is on a line by itself?
  1. Some of modified docstrings are also now have lines longer than 80 cols.  Can you wrap these?
  1. Some of the functions with modified docstrings have a body of `pass`.  The `pass` is redundant.  Can you delete it?
  1. The new text in the `say` docstring adds a second line to the `channel` section.  Can you indent that second line by 4 spaces (see, eg, the `length` docs in that same docstring)?
  1. There are some _simple suites_ on lines that this patch is modifying.  eg:
```

if channel[0] not in '&#!+': channel = '#' + channel
```
  The coding standard forbids simple suites.  Can you reformat these as:
```

if channel[0] not in '&#!+':
    channel = '#' + channel
```

That's all.  Thanks for working on this. :)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Looks good to me.  I'll apply it.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [27242]) Apply words-protocol-irc-v2.patch

Author: jesstess, ivank
Reviewer: exarkun
Fixes: [#3968](https://github.com/twisted/twisted/issues/3968)

Correct the documentation for a number of `IRCClient` methods, particularly
with respect to their treatment of channel names without channel prefixes.
Also clean up some nearby formatting to comply with the coding standard.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
